src/java.base/share/classes/jdk/internal/classfile/impl/StackMapDecoder.java

```diff
@@ -63,31 +63,30 @@ public class StackMapDecoder {
     static List<VerificationTypeInfo> initFrameLocals(MethodModel method) {
         return initFrameLocals(method.parent().orElseThrow().thisClass(),
                 method.methodName().stringValue(),
-                method.methodType().stringValue(),
+                method.methodTypeSymbol(),
                 method.flags().has(AccessFlag.STATIC));
     }
 
-    public static List<VerificationTypeInfo> initFrameLocals(ClassEntry thisClass, String methodName, String methodType, boolean isStatic) {
-        var mdesc = MethodTypeDesc.ofDescriptor(methodType);
+    public static List<VerificationTypeInfo> initFrameLocals(ClassEntry thisClass, String methodName, MethodTypeDesc methodType, boolean isStatic) {
         VerificationTypeInfo vtis[];
         int i = 0;
         if (!isStatic) {
-            vtis = new VerificationTypeInfo[mdesc.parameterCount() + 1];
+            vtis = new VerificationTypeInfo[methodType.parameterCount() + 1];
             if ("<init>".equals(methodName) && !ConstantDescs.CD_Object.equals(thisClass.asSymbol())) {
                 vtis[i++] = SimpleVerificationTypeInfo.ITEM_UNINITIALIZED_THIS;
             } else {
                 vtis[i++] = new StackMapDecoder.ObjectVerificationTypeInfoImpl(thisClass);
             }
         } else {
-            vtis = new VerificationTypeInfo[mdesc.parameterCount()];
+            vtis = new VerificationTypeInfo[methodType.parameterCount()];
         }
-        for(var arg : mdesc.parameterList()) {
-            vtis[i++] = switch (arg.descriptorString()) {
-                case "I", "S", "C" ,"B", "Z" ->  SimpleVerificationTypeInfo.ITEM_INTEGER;
-                case "J" -> SimpleVerificationTypeInfo.ITEM_LONG;
-                case "F" -> SimpleVerificationTypeInfo.ITEM_FLOAT;
-                case "D" -> SimpleVerificationTypeInfo.ITEM_DOUBLE;
-                case "V" -> throw new IllegalArgumentException("Illegal method argument type: " + arg);
+        for(var arg : methodType.parameterList()) {
```

Comment:
`MethodTypeDesc::parameterList` copies the internal `ClassDesc[]`, so desugaring loop from `0` to `mdesc.parameterCount()` might be a small improvement for this and a few other cases.

Reply:
I plan to share a backing list in #13186, with API additions so users can avoid copying their input immutable parameter list as well. I see no reason to use a direct array, for MTD is not passed to and used by the VM, unlike MethodType.

Reply:
Good, I think it makes perfect sense for the MTD parameter list to be immutable and shareable without copying.

Reply:
Thank you for the review.
Yes, I'm looking forward to see it together with Constants API improvements in one benchamark ;)

Reply:
Since you are available, can you help review my CSR request at https://bugs.openjdk.org/browse/JDK-8306699 as requested by Joe Darcy so we can get this improvement into release 21?